commands/proto.go

```diff
@@ -199,28 +314,22 @@ func (cmd *GenerateProtoCmd) addDataProperties(output, rmk string) string {
 			resourceTypeList = append(resourceTypeList, r.Value)
 			fieldNumber, cmd.i = util.GetFieldNumber(cmd.DeterministicFieldNumbers, cmd.i, resourceTypeList...)
 
-			// Make name and id mandatory
-			// TODO(oxisto): somehow extract this out of the ontology file itself which fields have constraints
-			if r.Value == "name" || r.Value == "id" {
-				opts = "[ (buf.validate.field).required = true ]"
-			}
+			optsOutput = cmd.emitPropertyOptions(r)
 
 			// Add data property comment if available
 			if r.Comment != "" {
 				output += fmt.Sprintf("\n\t// %s", r.Comment)
 			}
 
-			output += fmt.Sprintf("\n\t%s %s = %d%s;", r.Typ, util.ToSnakeCase(r.Value), fieldNumber, opts)
+			output += fmt.Sprintf("\n\t%s %s = %d%s;", r.Typ, util.ToSnakeCase(r.Value), fieldNumber, optsOutput)
 		}
 	}
 
 	return output
 }
 
-func (cmd *GenerateProtoCmd) addClassHierarchy(output, rmk string) string {
-	for _, typ := range cmd.getResourceTypeList(cmd.preparedOntology.Resources[rmk]) {
-		output += fmt.Sprintf("\toption (resource_type_names) = \"%s\";\n", typ)
-	}
+func (cmd *GenerateProtoCmd) addClassHierarchy(output, iri string) string {
```

Comment:
Do we really need this method?

Reply:
I just wanted to keep it in case we want to extend it.